### DISCUSSION TITLE
🧩 FEATURE REQUEST: Adding a navigation shortcut

### DISCUSSION THREAD

**Discussion by Kvaibhav01** [dailydotdev/daily#1017] [Feature Requests]
## Feature Request

This is a feature request to add a navigation shortcut or quick nav. This is primarily be used to do seamless tasks in the app/extensions quickly.

### Description

There can be some shortcuts to the app/extension. Like we can have <kbd>Cmd + S</kbd> to open sidebar, <kbd>Cmd + C</kbd> to comment etc.

### Solution

I recently got to know about this awesome library related to this called **[Scoutbar](https://www.scoutbar.co)**.  A user can navigate the app/extension with a keyboard only for quick tasks like:

1. Opening the navigation shortcut with <kbd>Cmd + K</kbd>.
2. There we can see a list of quick actions we can do like opening profile, settings, sidebar, searching for a post, opening comments or post page, opening the Github of Daily.
3. To close this we can use the <kbd>ESC</kbd> key.

### Alternate Solutions

N/A

**Comment by idoshamun** [comment]
Hi @Kvaibhav01,

This is a great suggestion. I'll keep you posted :)

**Comment by FrancescoXX** [comment]
thank you for this suggestion!

**Comment by h0rasan** [comment]
Bh

**Comment by jasiralavi** [comment]
Copied from https://github.com/dailydotdev/daily/discussions/1415

It would be super cool to have full navigation control using keyboard. I'm surprised devs here have not prioritized this, this being a platform for devs.

- Use arrow keys to navigate My Feed. Use tab after that to srcoll though the buttons after that (already works)
- Space or Enter to open the post (works now with enter)
- Shortcuts to Bookmark, save, etc from the feed as well as from when the news item is open. For example:
  → Tab+B to Bookmark,
  → Tab+C to copy link,
  → Tab+U or T++ to Upvote,
  → Tab+D or Tab+- to Downvote,
  → Tab+K for search,
  → Tab+N for New Post,
  → Tab+M to show other menu items (the actions list),
  → Tab+T to toggle navigation focus between the feed and the left sidebar,
  → Tab+H to open Help and shortcuts cheat-sheet, etc.
- It can be Ctrl / Tab / Alt etc. But let users configure it. This is important since different OS and browsers behave different. I use firefox on Linux and have both search-box and address-box both on, so for me Ctrl+K takes my cursor focus to the browser search-box, not to the dev.daily search

I've suggested Tab+ shortcuts as it doesn't conflict with os/app level shortcuts. It's used by many webapps now. e.g. Asana https://help.asana.com/s/article/keyboard-shortcuts?language=en_US

**Comment by AyushmanOfficial** [comment]
So its the command palette :D
